compiler-core/src/javascript/pattern.rs

```diff
@@ -50,32 +50,104 @@ pub enum BitArrayTailSpreadType {
     Bytes,
 }
 
-struct Offset {
-    bits: usize,
+#[derive(Debug, Clone)]
+pub struct OffsetBits<'a> {
+    constant: usize,
+    variables: Vec<Document<'a>>,
+    extra: Option<EcoString>,
+}
+
+impl<'a> OffsetBits<'a> {
+    fn increment(&mut self, size: BitArraySize<'a>) {
+        match size {
+            BitArraySize::Literal(size) => self.constant += size,
+            BitArraySize::Variable(name) => self.variables.push(name),
+        }
+    }
+
+    fn is_whole_number_of_bytes(&self) -> bool {
+        self.variables.is_empty() && self.extra.is_none() && self.constant % 8 == 0
+    }
+
+    fn divide(mut self, by: usize) -> Self {
+        if self.variables.is_empty() {
+            self.constant /= by;
+        } else {
+            self.extra = Some(eco_format!(" / {by}"));
```

Comment:
This seems to be the only time that `extra` is used, so it seems underspecified for it to be a string. This method also overwrites the previous division in one state, but it divides again in the other state. The behaviour should always be the same.